src/app/hooks.ts

```diff
@@ -0,0 +1,6 @@
+import { useDispatch } from 'react-redux';
+import { AppDispatch, RootState } from './store';
+import { useSelector } from 'react-redux';
+
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
```

Comment:
The `useDispatch.withTypes<AppDispatch>()` is incorrect. You should use `useDispatch<AppDispatch>()` instead to correctly type the dispatch function.